src/main/kotlin/com/jetbrains/snakecharm/lang/psi/SmkFileReference.kt

```diff
@@ -158,17 +179,26 @@ class SmkIncludeReference(
     }
 }
 
+//can be in any directory
+//version 6.5.1
 class SmkConfigfileReference(
         element: SmkArgsSection,
         textRange: TextRange,
         stringLiteralExpression: PyStringLiteralExpression,
         path: String
 ) : SmkFileReference(element, textRange, stringLiteralExpression, path) {
-    override fun getVariants() =  collectFileSystemItemLike() {
+    override fun getVariants() =  collectFileSystemItemLike(
+        onlyFromParentFolder = false,
+        isFromCurrentDir = false
+    ) {
         isYamlFile(it)
     }
+
+    override fun resolve(): PsiElement? = findPathToResolve(searchFromRoot = true)
```

Comment:
At the moment we have 3 flags: `onlyFromParentFolder`, `isFromCurrentDir`, `searchFromRoot`. At the moment the difference is not clear and no documentation available. I assume that `onlyFromParentFolder`, `isFromCurrentDir` is same thing. Could you refactor this code:
* convert this fields to `SmkFileReference` class abstract variables
* rename them to more clear names + add documentation. I suppose good names could be `searchRelativelyToContentRoot`, `searchRelativelyToCurrentFolder`, where current folder is parent folder of current snakemake file, and content root is the snakemake project working directory

Reply:
We have for example "conda:" field,  where we search file from root, but path is making from current folder:
"../envs/snpsift.yaml"
Therefore, we need 2 flags to search for a file and build a path

Reply:
I don't understand a bit what an abstract variable is, so for now I have made it the usual fields of the class

Reply:
I meant of abstract property:
```kotlin
abstract class SmkFileReference(..) {
   abstract val  searchRelativelyToCurrentFolder: Boolean
}
class Foo(): SmkFileReference(..) {
  override val searchRelativelyToCurrentFolder = True
} 
```
but your choice (private property + initialization via constructor) is also a good choice. Let's leave it as is.

Reply:
> We have for example "conda:" field
Discussion moved to the new PR.